psi4/share/psi4/scripts/test_threading.py

```diff
@@ -108,19 +119,87 @@ def test_psithon(args):
 
     rat1 = times[threads[0]] / times[threads[-1]]
     print("   Psi4@n%d : Psi4@n%d ratio (want ~%d): %.2f" % (threads[0], threads[-1], threads[-1], rat1))
+    if args.passfail:
+        #assert math.isclose(rat1, threads[-1], rel_tol=0.6), 'Psithon speedup {} !~= {}'.format(rat1, threads[-1])
+        assert rat1 > 1.3, '{} Psithon speedup {} !~= {}'.format(label, rat1, threads[-1])
+
 
+def test_plugin_dfmp2(args):
 
-def print_math_ldd():
-    import sys
-    import subprocess
+    inputdat = """
+import %s
 
-    import psi4
-    p4core = psi4.__file__[:-11] + 'core.so'
+memory 2 gb
+
+molecule {
+0 1
+C     0.0000000    0.0000000    1.0590353
+C     0.0000000   -1.2060084    1.7576742
+C     0.0000000   -1.2071767    3.1515905
+C     0.0000000    0.0000000    3.8485751
+C     0.0000000    1.2071767    3.1515905
+C     0.0000000    1.2060084    1.7576742
+H     0.0000000    0.0000000   -0.0215805
+H     0.0000000   -2.1416387    1.2144217
+H     0.0000000   -2.1435657    3.6929953
+H     0.0000000    0.0000000    4.9301499
+H     0.0000000    2.1435657    3.6929953
+H     0.0000000    2.1416387    1.2144217
+--
+0 1
+C    -1.3940633    0.0000000   -2.4541524
+C    -0.6970468    1.2072378   -2.4546277
+C     0.6970468    1.2072378   -2.4546277
+C     1.3940633    0.0000000   -2.4541524
+C     0.6970468   -1.2072378   -2.4546277
+C    -0.6970468   -1.2072378   -2.4546277
+H    -2.4753995    0.0000000   -2.4503221
+H    -1.2382321    2.1435655   -2.4536764
+H     1.2382321    2.1435655   -2.4536764
+H     2.4753995    0.0000000   -2.4503221
+H     1.2382321   -2.1435655   -2.4536764
+H    -1.2382321   -2.1435655   -2.4536764
+}
+
+set {
+  scf_type df
+  mp2_type df
+  basis aug-cc-pvdz
+  freeze_core true
+}
+
+e, wfn = energy('plugdfmp2', return_wfn=True)
+compare_values(-1.6309450762271729, wfn.get_variable('MP2 CORRELATION ENERGY'), 5, 'df-mp2 energy')  # aug-cc-pvdz
+#compare_values(-1.5720781831194317, wfn.get_variable('MP2 CORRELATION ENERGY'), 5, 'df-mp2 energy')  # cc-pvdz
+""" % (args.module)
+
+    tfn = '_dfmp2_plugin_thread_test_input_psi4_yo'
+    with open(tfn + '.in', 'w') as fp:
+        fp.write(inputdat)
+
+    run_psithon_inputs(args, tfn=tfn, label='Plugin dfmp2')
+
+
+def print_math_ldd(args):
+
+    module, sharedlibrary = args.module.split('/')
+    mod = importlib.import_module(module)
+    modcore = mod.__file__[:-11] + sharedlibrary
 
     if sys.platform.startswith('linux'):
-        cmd = """ldd -v {} | grep -e ':' -e 'mkl' -e 'openblas' -e 'iomp5' -e 'gomp'""".format(p4core)
+        cmd = """ldd -v {} | grep -e ':' -e 'mkl' -e 'openblas' -e 'iomp5' -e 'gomp'""".format(modcore)
```

Comment:
Im surprised, but as far as I can tell `ctypes` doesn't let you do this. Is it worth it to do platform localization `otool -L` on OSX here?

Reply:
harder for Mac b/c `otool` doesn't separate deps like `ldd -v` does. will try when I get to downstream, though. I can't figure out what `ctypes` has to do with this?

Reply:
`ctypes` has plenty of inspection capabilities built into it when loading shared objects. Finding out a SO's rpath and and links seems like it would be in its domain.